### PR TITLE
Papf2f

### PR DESCRIPTION
**Pull request by yuliumitre**
This PR creates the Face-to-Face encounter form for PAP.

Please use the branch of test-ehr : https://github.com/HL7-DaVinci/test-ehr/pull/23

### CODE REVIEW

File: HomeOxygenTherapy/R4/files/HomeOxygenTherapyLabPrepopulation-0.1.0.cql [hunk 1]
```diff
@@ -17,6 +17,8 @@ context Patient
 
 define Today: Today()
 
+define "True": True
```
Comment (yunwwang) [inline HomeOxygenTherapy/R4/files/HomeOxygenTherapyLabPrepopulation-0.1.0.cql]: True or true?

File: PositiveAirwayPressureDevices/R4/files/PositiveAirwayPressureDevicesPrepopulation-0.1.0.cql [hunk 2]
```diff
@@ -47,6 +55,33 @@ define PapDeviceRequested:
...
+define "RDI": HighestObservation(WithUnit(Verified(ObservationLookBack([Observation: "RDI_Codes"], 3 months)), '/h'))
+
+define "AllAlergy": ActiveAllergy([AllergyIntolerance])
+
+define "AllAllergyWithCode": 
+  distinct(flatten(
+    [AllergyIntolerance] A
+      let Codings: 
+        (A.code.coding) CODING where CODING.system.value in {
+          'http://snomed.info/sct'
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/files/PositiveAirwayPressureDevicesPrepopulation-0.1.0.cql]: Allergy should use RxNorm codes

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json [hunk 3]
```diff
@@ -0,0 +1,1760 @@
...
+          "type": "string",
+          "required": false
+        },
+        {
+          "linkId": "2.3",
+          "text": "Middle Initial",
+          "type": "string",
+          "required": false
+        }
+      ]
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json:216]: missing provider NPI

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json [hunk 4]
```diff
@@ -0,0 +1,1760 @@
...
+                "expression": "\"PositiveAirwayPressureDevicePrepopulation\".True"
+              }
+            }
+          ]
+        },
+        {
+          "linkId": "3.3",
+          "text": "Order purpose: ",
+          "type": "group",
+          "item": [
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json:262]: should have enableWhen: 3.2 = true

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json [hunk 5]
```diff
@@ -0,0 +1,1760 @@
...
+          ]
+        },
+        {
+          "linkId": "3.4",
+          "text": "If No, purpose of the encounter: ",
+          "type": "text",
+          "required": false,
+          "enableWhen": [
+            {
+              "question": "3.1",
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json]: 3.2?

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json [hunk 6]
```diff
@@ -0,0 +1,1760 @@
...
+              "url": "http://hl7.org/fhir/StructureDefinition/cqf-expression",
+              "valueExpression": {
+                "language": "text/cql",
+                "expression": "\"PositiveAirwayPressureDevicePrepopulation\".OsaDiagnosisExists"
+              }
+            }
+          ],
+          "linkId": "4.1",
+          "type": "boolean",
+          "text": "Obstructive Sleep Apnea (G47.33)"
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json:352]: this is required

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json [hunk 7]
```diff
@@ -0,0 +1,1760 @@
...
+              "url": "http://hl7.org/fhir/StructureDefinition/cqf-expression",
+              "valueExpression": {
+                "language": "text/cql",
+                "expression": "\"PositiveAirwayPressureDevicePrepopulation\".OtherDiagnoses"
+              }
+            }
+          ],
+          "linkId": "4.2",
+          "text": "Other diagnoses:",
+          "type": "open-choice",
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json:366]: this is required

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json [hunk 8]
```diff
@@ -0,0 +1,1760 @@
...
+    {
+      "linkId": "8",
+      "type": "group",
+      "text": "Ability of being trained",
+      "item": [
+        {
+          "linkId": "8.1",
+          "text": "Is the patient capable of being trained to use the device in an appropriate manner?",
+          "type": "boolean",
+          "extension": [
```
Comment (yunwwang) [inline PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json:530]: this is required

File: PositiveAirwayPressureDevices/R4/files/PositiveAirwayPressureDevicesPrepopulation-0.1.0.cql (hunk 2 above)
Comment (yuliumitre) [reply PositiveAirwayPressureDevices/R4/files/PositiveAirwayPressureDevicesPrepopulation-0.1.0.cql]: There are multiple types of allergy. We probably should include all of them. RxNorm is for drug allergy. 

https://www.hl7.org/fhir/v2/0127/index.html

Here are more granulated codes for animal allergies:
https://phinvads.cdc.gov/vads/ViewCodeSystemConcept.action?oid=2.16.840.1.113883.6.96&code=717234006

Probably need to use a value set.

File: PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json (hunk 5 above)
Comment (yuliumitre) [reply PositiveAirwayPressureDevices/R4/resources/Questionnaire-R4-PositiveAirwayPressureDevicesFaceToFace.json]: Good catch 👍

File: PositiveAirwayPressureDevices/R4/files/PositiveAirwayPressureDevicesPrepopulation-0.1.0.cql (hunk 2 above)
Comment (yunwwang) [reply PositiveAirwayPressureDevices/R4/files/PositiveAirwayPressureDevicesPrepopulation-0.1.0.cql]: In CDE: ALL: Allergies (include RxNorm if Known). My understanding is that only include medication allergies. Include RxNorm code if known.

**Review by yunwwang — APPROVED** [review]
Look good

**Pull request merged**